src/Darwin/OSLog.cs

```diff
@@ -0,0 +1,121 @@
+//
+// OSLog.cs: Bindings to the os_log(3) API from Apple
+//
+// Authors:
+//  William Kent
+//
+// Copyright 2019, Xamarin Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using ObjCRuntime;
+using System.Runtime.InteropServices;
+using System.Collections.Generic;
+
+namespace Darwin {
+	public class OSLog : IDisposable, INativeObject {
```

Comment:
You need to add availability attributes on the type so it will apply to all public members and the p/invokes (even private ones). Our introspection tests requires them to ensure we're not exposing unavailable symbols.

It's also helpful, from the IDE, to know the availability of an API. Figuring it out at runtime is much less fun :)